QEP-auxiliary-storage.rst

```diff
@@ -0,0 +1,178 @@
+an SQLite file.
+It only adds a step after actually saving the XML tree: including it in a Zip file. Symetrically during the loading, the first step would be to extract the XML file and pass it to the
+existing readXML methods.
+
+Auxiliary fields
+-------------------
+
+An auxiliary field is defined for a given layer. It is a new type of field: a new enum value for QgsFields::FieldOrigin (OriginAuxiliary). The code
+of QgsVectorLayer, QgsVectorLayerFeatureIterator and QgsVectorLayerEditBuffer will be changed to reflect this new type, so that they will appear in the list of fields of a layer (alongside native fields and virtual fields).
+
+Since an auxiliary field is conceptually a (left) join on an external database table, auxiliary fields may only be added to vector layers that have a **primary key**.
+
+Auxiliary fields are defined as a table in the central SQLite database.
+
+For instance, for a vector layer L with columns (ID, A, B, AF1, AF2) with native fields ID (primary key), A, B and auxiliary fields AF1 and AF2, there is
+an auxiliary table T with columns (ID, AF1, AF2). The auxiliary table is named after the name of the origin layer.
+
+A new row is added to the auxiliary table when a new row has been added to the origin vector layer and a data has been set (not null).
+
+A row is removed from the auxiliary table when the corresponding row has been removed from the origin vector layer.
+
+An auxiliary table is added when the first auxiliary field of a layer is created.
+
+An auxiliary table is removed and data are lost when the origin layer is closed. The closing of a layer should explicitly warn the user about the loss of auxiliary data.
+An auxiliary table is also removed when every auxiliary fields of a layer are removed.
+
+Auxiliary field as cache for expression evaluation
+--------------------------------------------------
+
+For data providers that do not support on-the-fly computation of values, virtual fields allow to augment a virtual layer by a field computed by an expression.
+
+For some particular use cases (computing geometries by expressions), such computations may be slow. Evaluating the expression that defines a virtual field is needed only when
+the fields it refers to change. We then propose to allow auxiliary fields to store values that are initially defined as expressions.
+
+An expression-defined auxiliary field can only be accessed for reading. The underlying fields refered to by the expression will be listed and when attributes of the vector layer
+is updated, linked expression-defined auxiliary fields will be updated.
+
+The value of an expression of complex type may need a serialization step before being cached. For instance, values of type QgsGeometry will be stored as WK(B|T).
+
+(Actually this cache mechanism for virtual fields is not specific to auxiliary fields and could be generalized to any tpyes of providers)
+
```

Comment:
This section appears after the label path QEP. I was concerned about possible performance issues.
But your remarks are right, thanks.
So ok to remove it or rework it as another (future) QEP for now